src/Chorus/UI/Clone/GetCloneFromInternetDialog.cs

```diff
@@ -261,11 +261,43 @@ private void _cancelButton_Click(object sender, EventArgs e)
 		}
 
 		private void OnDownloadClick(object sender, EventArgs e)
+		{
+			StartClone();
+		}
+
+		
+		/// <summary>
+		/// Starts a clone operation with the supplied information.
+		/// <param name="username">Username for Mercurial authentication</param>
+		/// <param name="password">Password for Mercurial authentication</param>
+		/// <param name="projectFolder">The parent directory to put the clone in</param>
+		/// <param name="projectName">Name of the project to clone</param>
+		/// <param name="projectUri">URI where the project can be found</param>
+		/// </summary>
+		public static GetCloneFromInternetDialog StartClone(string username, string password, string projectFolder, string projectName, Uri projectUri)
```

Comment:
Nitpicking: This dialog seems kind of a funny place for this method. I'd consider `GetSharedProjectModel` or even a new class.

Reply:
If you'd like me to move it, let me know where. (It doesn't matter too much to me.) I put it here, because it needs the model and dialog, both of which are present here. But its being static now makes separation easier